File: app/src/main/java/com/example/jetpackcomponentsapp/view/holder/BaseViewHolder.java
package com.example.jetpackcomponentsapp.view.holder;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.jetpackcomponentsapp.model.CustomModel;
import com.example.jetpackcomponentsapp.view.CustomListeners;

abstract class BaseViewHolder extends RecyclerView.ViewHolder {

    /**Main */
    private final CustomListeners listeners;

    public BaseViewHolder(CustomListeners listeners, @NonNull View itemView) {
        super(itemView);
        this.listeners = listeners;
    }

    public Context getContext() {
        return itemView.getContext();
    }

    public CustomListeners getListener() {
        return listeners;
    }

    abstract public void bindDataToViewHolder(CustomModel item, int position);
}

File: app/src/main/java/com/example/jetpackcomponentsapp/view/holder/BaseViewHolder.kt
package com.example.jetpackcomponentsapp.view.holder

import android.content.Context
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.example.jetpackcomponentsapp.view.listeners.CustomListeners
import com.example.jetpackcomponentsapp.model.CustomHolderModel

abstract class BaseViewHolder : RecyclerView.ViewHolder {

    private val customListeners : CustomListeners

    constructor(view : View, customListeners : CustomListeners) : super(view) {
        this.customListeners = customListeners
    }

    protected fun getContext() : Context {
        return itemView.getContext()
    }

    protected fun getListener() : CustomListeners {
        return customListeners
    }

    abstract fun bindDataToViewHolder(model : CustomHolderModel?, position : Int)
}